bazel/repository_locations.bzl

```diff
@@ -1,32 +1,73 @@
+# Envoy dependencies may be annotated with the following attributes:
+#
+# use_category - list of the categories describing how the dependency is being used. This attribute is used
+#                for automatic tracking of security posture of Envoy's dependencies.
+#                Possible values are documented in the USE_CATEGORIES list.
+
+# NOTE: If a dependency use case is either dataplane or controlplane, the other uses are not needed
+# to be declared.
+USE_CATEGORIES = [
+    # This dependency is used in build process.
+    "build",
+    # This dependency is used for unit tests.
+    "test",
+    # This dependency is used in API protos.
+    "api",
+    # This dependency is used in processing downstream or upstream requests.
+    "dataplane",
+    # This dependency is used to process xDS requests.
+    "controlplane",
+    # This dependency does not handle untrusted data and is used for various utility purposes.
+    "other",
+]
+
+def _dependency_location(urls, sha256, use_category, strip_prefix = None):
```

Comment:
Thanks, this is great. Can you move this macro out of this file so that it looks closer to pure Python? The rationale is that we may want to consume this in Python scripts that don't understand Starlark primitives like `fail()`.

Reply:
OK. We could change it back to `dict` and do verification in bazel/repositories.bzl It may be a bit confusing but probably ok. What do you think?

Reply:
Surprised not to see a Python implementation of Starlark at https://github.com/bazelbuild/starlark/blob/master/users.md#implementations

Reply:
I've moved validation into repositories.bzl